File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int j =0,k=0;
        for(int i =0; i< nums.size();i++){
           if(nums[i]!=0){
            nums[j] = nums[i];
            j++; 
           }
           if(nums[i] == 0){
            k++;
           }
        }
         for(int i =0; i< k;i++){
            nums.pop_back();
         }
         for(int i =0; i< k;i++){
            nums.push_back(0);
         }

    }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
       
       for(int i=0;i<nums.length;i++){
          if(nums[i] == 0){
            int idx = i;
            while(idx<nums.length && nums[idx] == 0){
             idx++;
            }
            if(idx < nums.length && idx != i){
            nums[i] = nums[idx];
            nums[idx] = 0;
            }
          }

       }
    }
}